In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('RANDOM.csv')

In [6]:
df.head(5)

Time Event  Count  VALUE[IN]   ID  Unnamed: 5
0   0.0     A    1.0      0.000  0.0         NaN
1   1.0     B    1.0      0.096  1.0         NaN
2   2.0     B    2.0      0.833  2.0         NaN
3   3.0     B    3.0      0.947  3.0         NaN
4   4.0     B    4.0      0.035  4.0         NaN

In [7]:
# x is our U[0,1] and a subset of VALUE[IN]
x = df['VALUE[IN]'].values[1:-1]

In [11]:
n = len(x)
n

1000

In [9]:
x.mean()

0.49072699999999997

In [10]:
x.std()

0.28647867367572055

In [40]:
# 20 intervals each of length .05
k = 20

In [48]:
# between each entry is 1 bin (21 entries => 20 bins)
bins = np.arange(k+1)/float(k)
bins

array([ 0.  ,  0.05,  0.1 ,  0.15,  0.2 ,  0.25,  0.3 ,  0.35,  0.4 ,
        0.45,  0.5 ,  0.55,  0.6 ,  0.65,  0.7 ,  0.75,  0.8 ,  0.85,
        0.9 ,  0.95,  1.  ])

In [41]:
# determine which interval each x falls into
binplace = np.digitize(x, bins)
binplace

array([ 2, 17, 19,  1,  1,  2, 16, 12, 19, 16,  7,  4,  6, 16, 20,  3,  7,
        1,  6,  4, 18,  5, 19,  5, 19, 19,  7, 13, 11,  4, 19,  7, 20, 19,
       19,  2, 19, 18,  8,  3, 14, 12,  5,  4, 20,  7, 11, 15, 13, 10,  8,
       12,  7, 15,  3, 11,  2, 12,  4, 14, 19,  7, 13, 10,  2,  3, 12,  2,
        3,  9,  5,  5, 12,  2, 15, 12,  8,  6, 18,  4,  9, 13, 18, 20, 11,
       15, 14,  5,  7,  9, 16, 11,  6,  8,  6, 13,  3,  9, 15,  1,  1,  9,
       13, 10, 14,  2, 14,  1, 16, 15, 19, 14,  7,  4, 16, 20,  7,  9,  2,
        3,  6, 16,  1, 10, 16, 15,  7, 14,  4,  1, 18, 14, 18,  5, 18, 10,
        7,  6, 17, 13, 18, 11, 13, 19, 19,  5, 16, 14, 11, 15, 10, 15, 18,
        1,  3,  7,  6, 15,  2, 19, 13, 12,  2,  3,  6,  4,  3,  5, 15,  7,
       11,  7, 18,  7,  6,  3, 20, 19,  1, 16, 18, 13,  9, 11, 12,  5, 12,
       13, 17,  7, 15,  1, 18, 16, 18,  9, 13,  4, 11, 18, 10, 15,  7,  6,
       19, 13, 11, 14, 18,  4, 12,  5,  6, 15,  2, 13, 18,  4, 10, 14,  2,
       20, 15, 17,  4,  1

In [39]:
# how many observations in each interval
f = np.bincount(binplace)[1:]
f

array([53, 51, 41, 52, 59, 45, 65, 50, 58, 47, 45, 49, 51, 46, 54, 44, 41,
       49, 55, 45])

In [44]:
(f-n/k)**2/float(n/k)

array([ 0.18,  0.02,  1.62,  0.08,  1.62,  0.5 ,  4.5 ,  0.  ,  1.28,
        0.18,  0.5 ,  0.02,  0.02,  0.32,  0.32,  0.72,  1.62,  0.02,
        0.5 ,  0.5 ])

In [47]:
# chi-squared value
X2 = ((f-n/k)**2/float(n/k)).sum()
X2

14.52

$H_{0}$: LCG is U[0,1]

$H_{1}$: LCG is not U[0,1]

For $\alpha$ = .05 and k-1 degrees of freedom 

X2 = 14.52 $\geq$ $\chi^{2}_{1-\alpha, k-1}$ = 10.12

therefore, we reject $H_{0}$, the LCG is not U[0,1]

Seed was 12345